src/app/public/modules/file-attachment/file-drop-control.component.spec.ts

```diff
@@ -0,0 +1,2957 @@
+
+      fileDropControl.validateFn = function (file: any) {
+        if (file.file.name.indexOf('w') === 0) {
+          return errorMessage;
+        }
+      };
+
+      fixture.detectChanges();
+
+      FileDropTestHelper.setupStandardFileChangeEvent(fixture);
+
+      expect(filesChangedActual.rejectedFiles.length).toBe(1);
+      expect(filesChangedActual.rejectedFiles[0].file.name).toBe('woo.txt');
+      expect(filesChangedActual.rejectedFiles[0].file.size).toBe(2000);
+      expect(filesChangedActual.rejectedFiles[0].errorType).toBe('validate');
+      expect(filesChangedActual.rejectedFiles[0].errorParam).toBe(errorMessage);
+
+      expect(filesChangedActual.files.length).toBe(1);
+      expect(filesChangedActual.files[0].url).toBe('url');
+      expect(filesChangedActual.files[0].file.name).toBe('foo.txt');
+      expect(filesChangedActual.files[0].file.size).toBe(1000);
+
+      expect(componentInstance.attachment.value).toEqual({
+        files: [
+          {
+            file: {
+              name: 'foo.txt',
+              size: 1000,
+              type: 'image/png'
+            },
+            url: 'url'
+          }
+        ],
+        links: undefined
+      });
+    });
+
+    it('should allow the user to specify accepted types', () => {
+      let filesChangedActual: SkyFileDropChange;
+
+      fileDropControl.filesUploaded.subscribe(
+        (filesChanged: SkyFileDropChange) => filesChangedActual = filesChanged);
+
+      fileDropControl.acceptedTypes = 'image/png,image/tiff';
+
+      fixture.detectChanges();
+
+      FileDropTestHelper.setupStandardFileChangeEvent(fixture);
+
+      expect(filesChangedActual.rejectedFiles.length).toBe(1);
+      expect(filesChangedActual.rejectedFiles[0].file.name).toBe('woo.txt');
+      expect(filesChangedActual.rejectedFiles[0].file.size).toBe(2000);
+      expect(filesChangedActual.rejectedFiles[0].errorType).toBe('fileType');
+      expect(filesChangedActual.rejectedFiles[0].errorParam).toBe(fileDropControl.acceptedTypes);
+
+      expect(filesChangedActual.files.length).toBe(1);
+      expect(filesChangedActual.files[0].url).toBe('url');
+      expect(filesChangedActual.files[0].file.name).toBe('foo.txt');
+      expect(filesChangedActual.files[0].file.size).toBe(1000);
+
+      expect(componentInstance.attachment.value).toEqual({
+        files: [
+          {
+            file: {
+              name: 'foo.txt',
+              size: 1000,
+              type: 'image/png'
+            },
+            url: 'url'
+          }
+        ],
+        links: undefined
+      });
+    });
+
+    it('should reject a file with no type when accepted types are defined', () => {
+      let filesChangedActual: SkyFileDropChange;
+
+      fileDropControl.filesUploaded.subscribe(
+        (filesChanged: SkyFileDropChange) => filesChangedActual = filesChanged);
+
+      fileDropControl.acceptedTypes = 'image/png,image/tiff';
+
+      fixture.detectChanges();
+
+      let files = [
```

Comment:
It looks like some of these `let`s can be `const`s.

Reply:
Uhhh. Probably lol. I'll look through them. Most tests were taken straight from the existing file drop tests.

Reply:
*wipes sweat* - Done